packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2729,6 +2810,28 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     _lastBottomViewInset = WidgetsBinding.instance.window.viewInsets.bottom;
   }
 
+  Future<void> _performSpellCheck(final String text) async {
+    try {
+      final Locale? localeForSpellChecking = widget.locale ?? Localizations.maybeLocaleOf(context);
+      final List<SuggestionSpan>? spellCheckResults = await
+        _spellCheckConfiguration!.spellCheckService!.fetchSpellCheckSuggestions(localeForSpellChecking as Locale, text);
+
+      if (spellCheckResults == null) {
+        return;
```

Comment:
You don't need to clear the _spellCheckResults or anything like that in this case?

Reply:
This case is hit whenever a spell check request was canceled. We still would want the text to get rendered with the current (possibly modified) spell check results, though. If there are no spell check results, the result is not null, it's just an empty list, so this case does not include that.